lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -168,6 +244,276 @@ def partition_pdf(
         LogTime("finish", point=True)
         return deformable_layout
 
+    def _partition_pdf_batched(
+        self,
+        file: BinaryIO,
+        threshold: float = 0.4,
+        use_ocr=False,
+        ocr_images=False,
+        ocr_tables=False,
+        extract_table_structure=False,
+        table_structure_extractor=None,
+        extract_images=False,
+        batch_size: int = 1,
+    ) -> List[List["Element"]]:
+        LogTime("partition_start", point=True)
+        with tempfile.NamedTemporaryFile(prefix="detr-pdf-input-") as pdffile:
+            with LogTime("write_pdf"):
+                data = file.read()
+                data_len = len(data)
+                pdffile.write(data)
+                data = b""
+                pdffile.flush()
+                logging.info(f"Wrote {pdffile.name}")
+            stat = os.stat(pdffile.name)
+            assert stat.st_size == data_len
+            assert os.path.isfile(pdffile.name), f"missing {pdffile.name}"
```

Comment:
Why not use the result from stat() instead of causing another one here?  I'd do `stat.S_ISREG(stat.st_mode)`

Reply:
It was an ordering thing, I wrote the isfile check first and later had issues so came back to do the stat. Stat is now irrelevant, all that matters is the thing exists of the right size and the stat checks the exists condition.